packages/revocation/src/implementations/revocationOffChain.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  Contract, ethers, Event, utils,
+} from 'ethers';
+import { EwSigner } from '@ew-did-registry/did-ethr-resolver';
+import { abi as RevocationOffChainAbi } from '../constants/RevocationRegistryOffChain.json';
+
+export class RevocationOffChain {
+  private _revocationRegistryOffChain: Contract;
+
+  /**
+  * @param revoker - Entity which perform revocation
+  * @param addressOffChain - Address of the off chain claim's RevocationRegistry
+  */
+  constructor(
+    revoker: EwSigner,
+    addressOffChain: string,
+  ) {
+    this._revocationRegistryOffChain = new ethers.Contract(
+      addressOffChain,
+      RevocationOffChainAbi,
+      revoker,
+    );
+  }
+
+  /**
+  * Revokes the credential
+  * Returns true on success
+  *
+  * @param { string } credential - credential to be revoked
+  * @returns Promise<boolean>
+  */
+  async revokeRole(credential: string): Promise<boolean> {
+    try {
+      const tx = await this._revocationRegistryOffChain.revokeClaim(utils.namehash(credential));
+      const receipt = await tx.wait();
+      const event = receipt.events.find(
+        (e: Event) => (e.event === 'Revoked'),
+      );
+      if (!event) return false;
+    } catch (error) {
+      throw new Error(error);
+    }
+    return true;
+  }
+
+  /**
+  * checks the revocation details for a credential
+  * Returns the revokers and revocationTimeStamp for the revocations
+  *
+  * @param { string } credential - credential for which the status is to be checked
+  * @returns Promise<string>
+  */
+  async getRevocationDetail(credential: string): Promise<string[]> {
```

Comment:
I don't understand how the return type is just a `string[]` 🤔 . Isn't is an array of two string arrays? Like `Promise<Array<string[], string[]>>?

Reply:
I tested the code, consoled the output as well and it worked. AFAIK Promise takes on argument for one return type and same goes with Array as well. So it could be either Promise<string[]> or Promise<Array<string[]>>.